### PR TITLE
Add a warning message when value < min | value > max in sliderInput (…

### LINKED ISSUE #1552: sliderInput doesn't warn on invalid value
The docs for the sliderInput says about `value`: "A warning will be issued if the value doesn't fit between min and max." I don't get a warning when running an app with an out-of-range `value`.

The value properly gets rounded up/down to a valid setting, but without a warning.

### LINKED ISSUE #2910: sliderInput: documentation for `value` is wrong - no warning is issued when value is outside of [min, max]
The docs for `value` say:

> The initial value of the slider. A numeric vector of length one will create a regular slider; a numeric vector of length two will create a double-ended range slider. A warning will be issued if the value doesn't fit between min and max.

However, setting a value outside of min and max does not produce any messages anywhere.

### PR DESCRIPTION
**Pull request by ColinFay**
…will close #2910 and close #1552)

### CODE REVIEW

File: R/input-slider.R [hunk 1]
```diff
@@ -90,6 +90,15 @@ sliderInput <- function(inputId, label, min, max, value, step = NULL,
                     version = "0.10.2.2")
   }
 
+  if (value < min | value > max) {
```
Comment (wch) [inline R/input-slider.R:93]: I think this needs to handle the case where `value` has two values.

Comment (ColinFay) [reply R/input-slider.R]: Done via last commit ✅

**Comment by ColinFay** [comment]
I just notice #2910 is a duplicate of #1552, so both will be closed by this one

File: R/input-slider.R [hunk 2]
```diff
@@ -90,6 +90,15 @@ sliderInput <- function(inputId, label, min, max, value, step = NULL,
                     version = "0.10.2.2")
   }
 
+  if (min(value) < min | max(value) > max) {
```
Comment (cpsievert) [inline R/input-slider.R:93]:
```suggestion
  if (min(value) < min || max(value) > max) {
```

**Comment by cpsievert** [comment]
Ideally this warning should also be thrown in `updateSliderInput()`

**Comment by ColinFay** [comment]
@cpsievert This has been implemented in the last two commits :)

**Comment by cpsievert** [comment]
I think you also want `||` instead of `|`?

**Comment by cpsievert** [comment]
Also, I think it'd be worthwhile to wrap the warning in a reusable function (e.g. `assert_slider_value()`)

**Comment by ColinFay** [comment]
@cpsievert Yeah I didn't wrap it up inside a function as I wanted the warning to throw the name of the function it was called from, but I can bypass that by using a function parameter.
Let me know what you think about the last commit.

**Comment by cpsievert** [comment]
Looks good! Would you mind also calling `devtools::document()` to incorporate the change from https://github.com/rstudio/shiny/pull/3023

**Comment by ColinFay** [comment]
Sure, done ✅

File: R/input-slider.R [hunk 3]
```diff
@@ -253,6 +255,18 @@ findStepSize <- function(min, max, step) {
   }
 }
 
+# Throw a warning if ever `value` is not in the [`min`, `max`] range
+assert_slider_value <- function(value, min, max, fun){
+  if (isTRUE(min(value) < min || max(value) > max)) {
+    warning(noBreaks. = TRUE, call. = FALSE,
+            sprintf(
+              "In %s(): Trying to set a `value` outside of the [`min`, `max`] range (`value` = %s, `min` = %s, `max = %s). `value` will be set to `max`.",
```
Comment (cpsievert) [inline R/input-slider.R:263]:
```suggestion
              "In %s(): Trying to set a `value` outside of the [`min`, `max`] range (`value` = %s, `min` = %s, `max = %s). `value`(s) will be constrained to [`min`, `max`].",
```

**Comment by ColinFay** [comment]
@cpsievert integrated ✅ :)

**Comment by daattali** [comment]
Related to https://github.com/rstudio/shiny/issues/2909 , this PR will introduce warnings (not the ones you intended) when `value` is not numeric

```
Warning messages:
1: In min(value) : no non-missing arguments to min; returning Inf
2: In max(value) : no non-missing arguments to max; returning -Inf
```

**Comment by wch** [comment]
Looks like there are some issues with handling NULL `value` and `min`/`max`. This is relevant for `updateSliderInput`.

```R
# OK
assert_slider_value(1,2,3, "asdf")
#> Warning message:
#> In asdf(): Trying to set a `value` outside of the [`min`, `max`] range (`value` = 1, `min` = 2, `max = 3). `value`(s) will be constrained to [`min`, `max`]. 

# No message
assert_slider_value(1,2,NULL, "asdf")
#> Warning message:
#>  

assert_slider_value(NULL,2,3, "asdf")
#> Warning messages:
#> 1: In min(value) : no non-missing arguments to min; returning Inf
#> 2: In max(value) : no non-missing arguments to max; returning -Inf

assert_slider_value(NULL,NULL,NULL, "asdf")
#> Warning messages:
#> 1: In min(value) : no non-missing arguments to min; returning Inf
#> 2: In max(value) : no non-missing arguments to max; returning -Inf
```

I'll make changes so that these cases work.

**Comment by wch** [comment]
This is superseded by #3194, since I'm not able to push to the branch in this PR.

**Pull request closed without merging**